deid/dicom/pixels/clean.py

```diff
@@ -153,36 +153,34 @@ def clean(self, fix_interpretation=True, pixel_data_attribute="PixelData"):
                     # Each is a list with [value, coordinate]
                     mask_value, new_coordinates = coordinate_set
 
-                    for new_coordinate in new_coordinates:
-
-                        # Case 1: an "all" indicates applying to entire image
-                        if new_coordinate.lower() == "all":
-
-                            # no frames, just X, Y
-                            if len(self.original.shape) == 2:
-                                # minr, minc, maxr, maxc = [0, 0, Y, X]
-                                new_coordinate = [
-                                    0,
-                                    0,
-                                    self.original.shape[1],
-                                    self.original.shape[0],
-                                ]
-
-                            # (frames, X, Y, channel) OR (frames, X,Y)
-                            if len(self.original.shape) >= 3:
-                                new_coordinate = [
-                                    0,
-                                    0,
-                                    self.original.shape[2],
-                                    self.original.shape[1],
-                                ]
-
+                    # Case 1: an "all" indicates applying to entire image
+                    if new_coordinates.lower() == "all":
```

Comment:
This should be a list of coordinates, e.g., 
```
for new_coordinate in new_coordinates
```

Reply:
Is this a difference in how identified coordinates come in through a recipe vs the "from:field" functionality?  If we look at base [deid.dicom](https://github.com/pydicom/deid/blob/master/deid/data/deid.dicom#L297) recipe, instances of multiple regions being detected by a filter are identified by multiple coordinates lines.  Each one of these coordinates lines results in a separate "coordinate set".   With the loop in pace, this iterates through a single coordinate comma separated string and not through a list of coordinates.

Two of the test cases that I wrote specifically test for the multiple-region scenario, however, they are all recipe based tests.  I may need the sample ultrasound you mentioned to write some more tests if there's a difference in the way the coordinates are identified through the "from:Field" functionality

Reply:
There isn't a difference - whether the coordinate comes in via a from or a list, from, or all, it should still be rendered as a list, eg.,
```
['from:FieldName', 'all', '0,0,0,0']
```
So a recipe should be able to have more than one coordinate, possibly even in different formats. I'll send you the echo image right away!